frontend/components/Course/InstructorQueuePage/Queue.tsx

```diff
@@ -77,28 +118,88 @@ const Queue = (props: QueueProps) => {
         await mutate(queueId, { active: false });
     };
 
+    const pinInput = (
+        <div className="ui input">
+            <label
+                htmlFor="changePin"
+                style={{
+                    fontWeight: "bold",
+                    padding: "0.5rem 0.5rem 0.5rem 2rem",
+                }}
+            >
+                Pin:
+            </label>
+            <input
+                name="changePin"
+                ref={pinInputRef}
+                value={editingPin ? pinState : pin}
+                onChange={handlePinChange}
+                onKeyPress={(e: React.KeyboardEvent<HTMLInputElement>) => {
+                    editingPin && e.key === "Enter" && endPinEdit();
+                }}
+                style={{
+                    height: "2.7rem",
+                    color: editingPin ? "black" : "#B9B9BA",
+                    caretColor: editingPin ? "#75767F" : "transparent",
+                    borderColor: "#22242626",
+                    backgroundColor: editingPin ? "white" : "#F1F1F2",
+                    paddingRight: "2rem",
+                }}
+            />
+            <Button
```

Comment:
There seems to be a bug when you click the checkmark after you're done editing, it sets editing pin to true again
It only happens if you click right on the icon